shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h

```diff
@@ -159,6 +167,11 @@ NS_ASSUME_NONNULL_BEGIN
 
 /**
  Registration context for a single `FlutterPlugin`.
+
+ When called upon to register itself with a plugin registry, the plugin
+ is given a `FlutterPluginRegistrar` which provides the context for that
+ registration. The registrar protocol decouples the plugin from the registry
+ itself, narrowing the API depended upon by plugins.
  */
```

Comment:
You described the relationship of the plugin to the registry and to the registrar here. Perhaps also add a quick line for the relationship between the registry and the registrar themselves. i.e. how does registrar perform the registration on the registry? Does registrar has-a registry or vice versa?